File: py/recursion/pre-order.py
"""
# Definition for a Node.
class Node:
    def __init__(self, val=None, children=None):
        self.val = val
        self.children = children
"""

class Solution:
    def preorder(self, root: 'Node') -> List[int]:
        order = []
        def helper(node):
            order.append(node.val)
            for child in node.children:
                helper(child)
        if root:
            helper(root)
        return order

File: js/preOrder.js
/**
 * // Definition for a Node.
 * function Node(val, children) {
 *    this.val = val;
 *    this.children = children;
 * };
 */

/**
 * @param {Node|null} root
 * @return {number[]}
 */
var preorder = function(root) {
    if (root && root.children) {
        lst = [root.val]
        for(let i= 0;i<root.children.length;i++) {
            lst = lst.concat(preorder(root.children[i]))
        }
        return lst
    }
    else {
        return []
    }
};